### PR TITLE
Test tweaks for less order dependency

### PR DESCRIPTION
**Pull request by irwiss**
#### Summary
None

#### Purpose of change

Make tests less dependent on order

#### Describe the solution

Teleport player back to starting position after eoc test - clear_avatar() doesn't deal with omt teleport

Add missing clear_map() to 2 item_contents tests so that overflow/spill contents don't randomly fail if avatar is on obstacle

### CODE REVIEW

**Comment by andrei8l** [comment]
> Teleport player back to starting position after eoc test - clear_avatar() doesn't deal with omt teleport

This shouldn't be needed though. Tests (and their relevant code) are supposed to work regardless of position.

**Comment by irwiss** [comment]
This is the only test directly calling g->place_player_overmap, so either the specific test places avatar back (or roll back whatever side effects it causes) or we add it to clear_avatar which will make a lot of calls to g->place_player_overmap, test cleaning up after itself is cheaper i think

If you wanna test it out `./tests/cata_test --rng-seed 1686877286 "EOC_transform_radius","avatar diving"` should fail on master

**Comment by andrei8l** [comment]
> If you wanna test it out `./tests/cata_test --rng-seed 1686877286 "EOC_transform_radius","avatar diving"` should fail on master

That's a failure in the setup for `avatar diving`. Quick hack to fix it:
```diff
diff --git a/tests/map_helpers.cpp b/tests/map_helpers.cpp
index 8144a7cf68..435dfefb87 100644
--- a/tests/map_helpers.cpp
+++ b/tests/map_helpers.cpp
@@ -176,11 +176,11 @@ void build_water_test_map( const ter_id &surface, const ter_id &mid, const ter_i
     constexpr int z_surface = 0;
     constexpr int z_bottom = -2;
 
-    clear_map( z_bottom, z_surface );
+    clear_map( z_bottom - 1, z_surface + 1 );
 
     map &here = get_map();
-    for( const tripoint &p : here.points_in_rectangle( tripoint_zero,
-            tripoint( MAPSIZE * SEEX, MAPSIZE * SEEY, z_bottom ) ) ) {
+    for( const tripoint &p : here.points_in_rectangle( { 0, 0, z_surface + 1},
+            tripoint( MAPSIZE * SEEX, MAPSIZE * SEEY, z_bottom - 1) ) ) {
 
         if( p.z == z_surface ) {
             here.ter_set( p, surface );
@@ -188,6 +188,10 @@ void build_water_test_map( const ter_id &surface, const ter_id &mid, const ter_i
             here.ter_set( p, mid );
         } else if( p.z == z_bottom ) {
             here.ter_set( p, bottom );
+        } else if( p.z < z_bottom ) {
+            here.ter_set( p, t_rock );
+        } else if( p.z > z_surface ) {
+            here.ter_set( p, t_open_air );
         }
     }
 
diff --git a/tests/water_movement_test.cpp b/tests/water_movement_test.cpp
index 4c63ba419e..d05234f065 100644
--- a/tests/water_movement_test.cpp
+++ b/tests/water_movement_test.cpp
@@ -31,9 +31,9 @@ static void setup_test_lake()
 
 TEST_CASE( "avatar diving", "[diving]" )
 {
+    clear_avatar();
     setup_test_lake();
 
-    clear_avatar();
     Character &dummy = get_player_character();
     map &here = get_map();
     constexpr tripoint test_origin( 60, 60, 0 );
```

I must stress that OMT position should not affect test outcome. Occasionally some code only works at absolute origin and we want to catch those errors (example: https://github.com/CleverRaven/Cataclysm-DDA/pull/60586#discussion_r961305876)

**Comment by irwiss** [comment]
Yep that sounds reasonable, applied your fix
